service-base/src/main/java/org/eclipse/hono/service/credentials/CredentialsService.java

```diff
@@ -45,21 +45,47 @@
     /**
      * Add credentials to credentials registry.
      *
-     * TODO: params
-     * 
+     * @param tenantId The tenant the device belongs to.
+     * @param deviceId The ID under which the device is registered.
+     * @param type The type of credentials to add.
+     * @param authId The authID of the device to add credentials data for.
+     * @param credentialsObject A map containing keys and values that fulfill the credentials format of the credentials api.
+     *                  See <a href="https://www.eclipse.org/hono/api/Credentials-API/#credentials-format">Credentials Format</a> for details.
+     * @param resultHandler The handler to invoke with the result of the operation. If a device with the given ID does not
+     *         exist for the tenant, the <em>status</em> will be {@link HttpURLConnection#HTTP_PRECON_FAILED}.
+     *                      If credentials for the type and authId of the device are already existing, the <em>status</em> will be {@link HttpURLConnection#HTTP_CONFLICT}.
+     *         Otherwise the operation is successful and the status will be {@link HttpURLConnection#HTTP_CREATED}.
+     * @throws NullPointerException if any of the parameters is {@code null}.
      */
-    void addCredentials(String tenantId, String deviceId, String type, String authId, JsonObject otherKeys, Handler<AsyncResult<CredentialsResult>> resultHandler);
+    void addCredentials(String tenantId, String deviceId, String type, String authId, JsonObject credentialsObject, Handler<AsyncResult<CredentialsResult>> resultHandler);
```

Comment:
The *credentialsObject* already contains *deviceId*, *type* and *authId* (mandatory members of the Credentials Format JSON). Why do we need to specify them as explicit params? We need to parse the credentialsObject anyway in order to make sure that it complies with the spec, don't we?